tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp

```diff
@@ -761,27 +754,27 @@ ASTUnitRef ASTProducer::createASTUnit(SwiftASTManager::Implementation &MgrImpl,
   const InvocationOptions &Opts = InvokRef->Impl.Opts;
 
   SmallVector<FileContent, 8> Contents;
-  for (auto &File :
-       Opts.Invok.getFrontendOptions().Inputs.getInputFilenames()) {
-    bool FoundSnapshot = false;
-    for (auto &Snap : Snapshots) {
-      if (Snap->getFilename() == File) {
-        FoundSnapshot = true;
-        Contents.push_back(getFileContentFromSnap(Snap, File));
-        break;
-      }
-    }
-    if (FoundSnapshot)
-      continue;
-
-    auto Content = MgrImpl.getFileContent(File, Error);
-    if (!Content.Buffer) {
-      LOG_WARN_FUNC("failed getting file contents for " << File << ": " << Error);
-      // File may not exist, continue and recover as if it was empty.
-      Content.Buffer = llvm::MemoryBuffer::getNewMemBuffer(0, File);
-    }
-    Contents.push_back(std::move(Content));
-  }
+  Opts.Invok.getFrontendOptions().Inputs.forEachFilename(
+                                                         [](StringRef File) {
+                                                           bool FoundSnapshot = false;
+                                                           for (auto &Snap : Snapshots) {
+                                                             if (Snap->getFilename() == File) {
+                                                               FoundSnapshot = true;
+                                                               Contents.push_back(getFileContentFromSnap(Snap, File));
+                                                               break;
+                                                             }
+                                                           }
+                                                           if (FoundSnapshot)
+                                                             return;
+                                                           
+                                                           auto Content = MgrImpl.getFileContent(File, Error);
+                                                           if (!Content.Buffer) {
+                                                             LOG_WARN_FUNC("failed getting file contents for " << File << ": " << Error);
+                                                             // File may not exist, continue and recover as if it was empty.
+                                                             Content.Buffer = llvm::MemoryBuffer::getNewMemBuffer(0, File);
+                                                           }
+                                                           Contents.push_back(std::move(Content));
+                                                         });
```

Comment:
Just a drive-by.  This is pretty far indented.  Would be nice to just hoist this out and create a local variable for the lambda which you can pass as an argument.  It would significantly un-indent the code here.  There is at least one other instance of this.